python/pylibcudf/pylibcudf/scalar.pyx

```diff
@@ -111,6 +120,84 @@ cdef class Scalar:
         """
         return _from_py(py_val)
 
+    @classmethod
+    def from_numpy(cls, numpy_val):
+        """
+        Convert a NumPy scalar to a Scalar.
+
+        Parameters
+        ----------
+        numpy_val: numpy.generic
+            Value to convert to a pylibcudf.Scalar
+
+        Returns
+        -------
+        Scalar
+            New pylibcudf.Scalar
+        """
+        cdef DataType dtype
+        cdef unique_ptr[scalar] c_obj
+        cdef Scalar slr
+
+        import numpy as np
+
+        if isinstance(numpy_val, np.bool_):
```

Comment:
Can we singledispatch this?

Sorry to be a broken record on this issue. I have in general become very wary of long if/else chains of instance checks in cudf. It sucks that they matter, because they really shouldn't, but we have too many places in our library that use them because of the crazy number of different types of things we support as inputs to different functions. The result is that it's shockingly easy to wind up executing literally hundreds of instance checks -- some on types that implement `__instancecheck__` in Python -- and all of a sudden you're eating up milliseconds instead of nanoseconds before you even reach a real algorithm. In this case it's probably fine since hopefully we aren't calling `from_numpy` in a loop anywhere, but you never know when that might happen.

Reply:
No problem, totally understandable. My latest commits retooled this to use `singledispatch`